service/debugger.go

```diff
@@ -0,0 +1,402 @@
+package service
```

Comment:
I'd like to move this to service/debugger/debugger.go and name this package `debugger`. That way it's isolated into it's own package and clients can just call `debugger.New()`.